traitcuration/traits/datasources/zooma.py

```diff
@@ -0,0 +1,84 @@
+"""
+This module contains all the functionality that uses ZOOMA to retrieve mapping suggestions and create the suggested
+terms in the app's database
+"""
+import requests
+
+from ..models import Trait, MappingSuggestion, OntologyTerm, User
+from .ols import make_ols_query, get_ontology_id
+
+
+BASE_URL = "http://www.ebi.ac.uk/spot/zooma/v2/api"
+
+
+def get_zooma_suggestions():
+    """
+    Retrieves zooma mapping suggestions for all traits and creates terms and suggestions in the app's database for each
+    of them.
+    """
+    traits = Trait.objects.all()
+    for trait in traits:
+        results_list = get_suggestions_for_trait(trait)
+        for result in results_list:
+            term = create_local_term(result)
+            # If local term creation returns None, it means that this term already exists in the database
+            if term is not None:
+                create_mapping_suggestion(trait, term)
```

Comment:
Looks like this only adds suggestions but never removes the old ones. I think during the suggestion update cycle we can also remove old suggestions for a trait which are:
* Originally created by ZOOMA
* **Not** selected as an accepted suggestion for this trait

Reply:
(Otherwise the list of suggestions will just grow and grow and grow)

Reply:
So basically remove any preexisting suggestions for a trait made by ZOOMA and repopulate them in each cycle, right?

Reply:
Yes, correct. Also, thanks for noticing that the _previous_ mappings also shouldn't be removed, I didn't think of that at all.

Perhaps this removal could even be done by a single SQL command for all suggestions at once to improve performance (although this sort of optimisation is not necessary at this point).